dvc/repo/experiments/executor/base.py

```diff
@@ -76,6 +76,7 @@ class TaskStatus(IntEnum):
     FAILED = 4
     CANCELED = 5
     FINISHED = 6
+    FAILED_FINISHED = 7
```

Comment:
Not sure of any better name for this?

Reply:
Can you clarify what `FAILED_FINISHED` means?

One option would be to use bit flags (via `enum.IntFlag`) here instead of incrementing values. So then you could have something like
```
class TaskStatus(IntEnum):
    PENDING = 1
    PREPARING = 2
    RUNNING = 4
    SUCCESS = 8
    FAILED = 16
    CANCELED = 32
    FINISHED = 64
```

and then use `status = TaskStatus.FAILED | TaskStatus.FINISHED` to indicate a task has both states

Reply:
It seems like we might be running into problems due to equating celery task states to DVC exp states, which aren't necessarily the same thing.

Reply:
> class TaskStatus(IntEnum):
>     PENDING = 1
>     PREPARING = 2
>     RUNNING = 4
>     SUCCESS = 8
>     FAILED = 16
>     CANCELED = 32
>     FINISHED = 64

This will make the new status incompatible with versions. 

> It seems like we might be running into problems due to equating celery task states and DVC exp states, which aren't necessarily the same thing.

Actually here the task status is used for the `local` and `tmp_dir` executors to match the status of `celery` one.

BTW, For this PR, actually the status change is not a necessity, I'll delete related codes.